### ISSUE TITLE
Auto-handle torch.hub trust for Silero VAD ASR loading

### ISSUE THREAD

**Issue by Ame4th** [RachelForster/Shinsekai#209]
## 摘要 / Summary
ASR may fail on first use because torch.hub asks for trust confirmation when RealTimeSTT loads Silero VAD. Shinsekai should handle this non-interactively or provide a clear UI fallback.

希望在 ASR 初始化时自动处理 `torch.hub` 对 `snakers4/silero-vad` 的 trust repo 确认，避免后台线程无法交互导致 EOF，从而让用户无需手动在终端执行预信任命令。

当前群内排障结论显示：RealTimeSTT 加载 Silero VAD 时触发了 `torch.hub` 的“是否信任仓库”交互确认，但 Shinsekai 后台无法输入，导致初始化失败。对于普通用户来说，这会表现为 ASR 无法使用，且很难自行定位原因。

## 现象 / Problem
A user reported that ASR could not be used, and diagnosis pointed to a torch.hub trust prompt causing EOF in the backend. The current behavior can block first-time ASR usage for non-technical users.

用户反馈 ASR 无法使用。根据已有诊断，主因是 RealTimeSTT 在加载 Silero VAD 时触发 `torch.hub` 的信任确认，而 Shinsekai 后台无法对该交互提示输入确认，最终导致 EOF。

群内临时处理方式是让用户使用同一个 Python 3.10 环境手动执行：

```bash
python -c "import torch; torch.hub.load('snakers4/silero-vad', 'silero_vad', trust_repo=True)"
```

执行后再重启 Shinsekai、关闭旧聊天窗并重新启动聊天。但这个流程对普通用户不友好，建议在应用内自动处理或至少给出明确提示。

## 触发场景 / Trigger scenario
The reported scenario happens when starting chat or enabling microphone with the ASR backend that loads RealTimeSTT and Silero VAD. Exact environment details are still missing.

已知触发场景来自群内排障：

1. 使用会加载 RealTimeSTT / Silero VAD 的 ASR 路径。
2. 启动聊天或打开麦克风，触发 ASR 后台初始化。
3. 首次加载 Silero VAD 时，`torch.hub` 需要信任 `snakers4/silero-vad`。
4. 后台环境无法输入确认，初始化出现 EOF，ASR 无法继续使用。

更精确的环境、配置、日志和复现步骤仍需补充。

## 预期行为 / Expected behavior
Shinsekai should not require users to manually answer a hidden terminal prompt. First-time ASR initialization should either complete automatically or show an actionable UI message.

期望行为：

- 在 ASR 初始化过程中，自动处理 `torch.hub` 对 `snakers4/silero-vad` 的 trust repo 确认。
- 首次使用 ASR 时，不需要用户额外打开终端执行预加载命令。
- 如果因网络、权限、缓存损坏或依赖版本问题导致自动处理失败，应在 UI 中给出可理解的提示，而不是只让后台 EOF。
- 不应阻塞 UI 主线程；加载、预热或下载过程应保持后台执行。

## 实际行为 / Actual behavior
The backend cannot answer torch.hub's trust prompt, so initialization can end with EOF. Users may only see that ASR is unavailable or plugin loading failed.

实际表现为：

- ASR 初始化过程中触发 `torch.hub` 信任确认。
- Shinsekai 后台无法响应该交互输入。
- 初始化过程出现 EOF。
- 用户侧看到 ASR 无法使用，需要他人根据诊断包或日志定位问题。

另有群聊中提到 GPT-SoVITS 的 `127.0.0.1:9880` 连接被拒绝，但这是 TTS 服务未启动或端口不匹配的独立问题，不作为本 issue 的主要范围。

## 日志/截图 / Logs/Screenshots
No raw log file is included in this issue draft. The available diagnosis mentions RealTimeSTT, Silero VAD, torch.hub trust confirmation, and EOF.

当前 issue 草稿没有附带原始日志文件，仅有群内诊断结论：

- RealTimeSTT 加载 Silero VAD。
- `torch.hub` 触发 trust repo 交互确认。
- Shinsekai 后台无法输入确认。
- 最终导致 EOF。

另有截图信息显示插件列表中存在 `WhisperAsrPlugin（whisper_asr）`，状态为“已停用”，版本显示为 `—`。该截图只能说明当时界面状态，不能单独证明本 issue 的根因。

## 可能路径 / Possible path
The likely path is around ASR initialization and microphone startup. The implementation should avoid UI blocking and should only trust the known Silero VAD repository if automatic handling is accepted.

可能涉及以下位置，具体仍需以实际源码为准：

- `ui/chat_ui/mic_button.py`
  - 负责麦克风按钮、ASR 延迟初始化、ASR 启动，以及将后台线程中的 ASR 状态通知回 UI。
  - 如果需要把 EOF / trust repo 相关异常转成用户可理解的弹窗或提示，这里可能是接入点之一。

- `asr.asr_adapter`
  - `mic_button.py` 中导入了 `create_default_asr_adapter` 和 `get_asr_log`。
  - ASR 后端创建、RealTimeSTT 初始化或 Whisper ASR 适配逻辑可能在这里继续展开。

可考虑的实现方向：

1. 优先定位项目中是否有显式调用：

   ```python
   torch.hub.load('snakers4/silero-vad', 'silero_vad', ...)
   ```

   如果有，可评估在该调用处加入 `trust_repo=True` 或等价的非交互配置。

2. 如果 Silero VAD 加载隐藏在 RealTimeSTT 第三方库内部，项目侧可考虑在 ASR adapter 初始化前做一次受控预加载 / 预信任，确保 torch.hub 已记录信任状态后再启动 RealTimeSTT。

3. 如果自动处理失败，应捕获相关异常并通过现有 ASR 初始化失败 / 启动失败通知路径提示用户，例如提示检查网络、缓存权限，或手动执行预加载命令。

4. 预加载或下载过程不应放在 UI 主线程，避免启动聊天或点击麦克风时界面卡死。

## 风险与注意事项 / Risks and notes
Automatically trusting a torch.hub repository has security implications because torch.hub can execute remote code. The behavior should be scoped and logged clearly.

需要注意：

- `torch.hub` 加载 GitHub 仓库代码具有安全含义，自动 `trust_repo=True` 不应被泛化到任意仓库。
- 如果实现自动信任，建议只针对明确的 `snakers4/silero-vad` 仓库，并保留清晰日志。
- 自动信任不等于一定能下载成功；国内网络、离线环境、GitHub 访问失败仍可能导致初始化失败。
- 不同 torch 版本对 `trust_repo` 参数的支持和行为可能不同，需要兼容当前项目支持的依赖版本。
- torch.hub 缓存目录可能存在写权限、杀软拦截或缓存损坏问题；这些情况下仍需要可诊断的错误提示。

## 验收方式 / Acceptance criteria
A clean environment should be able to initialize the relevant ASR path without manual terminal confirmation. Failure cases should produce understandable UI messages and logs.

建议验收方式：

- 在全新用户环境或清空 torch.hub 缓存后，启动 Shinsekai。
- 选择会触发 RealTimeSTT / Silero VAD 的 ASR 路径。
- 启动聊天并打开麦克风。
- 不再需要用户在终端输入 `y`，也不需要手动执行 `torch.hub.load(..., trust_repo=True)`。
- 后台不再因 trust repo 交互确认而 EOF。
- 如果网络或权限导致加载失败，UI 能给出明确可理解的错误提示，日志中能看到具体失败原因。

## 相关 Issue / Related issues
Some existing issues are related to AI dependency installation or GitHub access, but they are not duplicates of this ASR trust prompt problem.

- #87：与 AI 依赖安装、torch 安装流程相关，但本 issue 的核心不是 torch 版本选择，而是 `torch.hub` 首次加载远程仓库时的 trust repo 交互问题，不是重复 issue。
- #35：与国内访问 GitHub API 稳定性相关。本 issue 可能受 GitHub 网络影响，但核心问题是后台无法处理 `torch.hub` 信任确认，不是重复 issue。

## 补充信息 / Additional information
More environment details and raw logs are needed to confirm the exact failing path and dependency versions.

待补充：

- Shinsekai 版本或提交号。
- 操作系统版本。
- Python 版本。
- torch 版本。
- RealTimeSTT 版本。
- 当前选择的 ASR 插件 / ASR 引擎配置。
- 完整 ASR 初始化日志或诊断包。
- 是否为首次运行，或是否清空过 torch.hub 缓存。
- torch.hub 缓存目录是否有写权限。
- 网络环境是否能访问 GitHub / `snakers4/silero-vad`。

## Labels
labels:
  - asr
  - bug
  - ux
  - p1

**Comment by kldsjfas** [comment]
查了下根因：RealtimeSTT 1.0.0 加载 Silero VAD 时没传 trust_repo，torch.hub 对未信任仓库会用 input() 询问，后台线程没有 stdin 就直接 EOF 了

修复提在插件仓库：RachelForster/Shinsekai-Whisper-Asr#1 —— 在 RealtimeSTTAdapter 创建 recorder 前先以 trust_repo=True 预载一次，把仓库写进 torch.hub 信任列表，之后 RealtimeSTT 自己的加载就不会再触发交互确认，用户也不用手动跑预信任命令了。在隔离 TORCH_HOME + 关闭 stdin 的环境下用真实 torch 复现并验证过（EOF 复现 → 预信任 → 原始调用方式不再询问）。

**Comment by RachelForster** [comment]
感谢，已修复这个问题
